src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -1484,7 +1484,8 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                 // If the whole program view contains a reference to a preallocated RuntimeType
                 // instance for this type, generate a reference to it.
                 // Otherwise, generate as zero to save size.
-                if (_type.GetFrozenRuntimeTypeNode(factory) is { Marked: true } runtimeTypeObject)
+                if (!_type.Type.IsCanonicalSubtype(CanonicalFormKind.Any)
+                    && _type.GetFrozenRuntimeTypeNode(factory) is { Marked: true } runtimeTypeObject)
```

Comment:
How is it getting marked in the first place?

Reply:
We assert in the `FrozenRuntimeTypeNode` constructor before we can check `Marked`

Reply:
I meant - what is the dependency chain that ends up marking FrozenRuntimeType for CanonicalSubtypes?

Reply:
> I meant - what is the dependency chain that ends up marking FrozenRuntimeType for CanonicalSubtypes?

Did the other conversation resolve this question? (It was RyuJIT. Now we guard it in JitInterface, but an assert in the FrozenRuntimeType ctor still feels useful. But the assert means we need to check for canonical here too because we can't create the node just to check if it's marked.)